consensus/parlia/parlia.go

```diff
@@ -701,10 +701,8 @@ func (p *Parlia) snapshot(chain consensus.ChainHeaderReader, number uint64, hash
 			}
 		}
 
-		// If we're at the genesis, snapshot the initial state. Alternatively if we have
-		// piled up more headers than allowed to be reorged (chain reinit from a freezer),
-		// consider the checkpoint trusted and snapshot it.
-		if number == 0 || (number%p.config.Epoch == 0 && (len(headers) > int(params.FullImmutabilityThreshold)/10)) {
```

Comment:
could be a problem, need to double check.

Reply:
it was a bug introduced by this PR: https://github.com/bnb-chain/bsc/pull/1381.
BSC will switch validator set at "epoch + len(ValidatorSet) +1", which is different from Ethereum.